HOWTO.md

```diff
@@ -0,0 +1,135 @@
+ <version>${openlr.version}</version>
+</dependency>
+```
+
+```xml
+<dependency>
+ <groupId>org.openlr</groupId>
+ <artifactId>proto</artifactId>
+ <version>${openlr.version}</version>
+</dependency>
+```
+
+#3. Map Implementation
+
+**Implement the following interfaces against the map you want to use**
+
+<ul>
+<li>openlr.map.Line</li>
+<li>openlr.map.Node</li>
+<li>openlr.map.MapDatabase</li>
+<li>openlr.map.loader.MapLoadParameter</li>
+<li>openlr.map.loader.OpenLRMapLoader</li>
+</ul> 
+
+#4 Load Map
+
+<p> We are using the sqlite database for the remaining part of the document </p>
+
+```java
+MapLoadParameter param = new DBFileNameParameter();
+param.setValue("path/to/map");
+OpenLRMapLoader loader = new SQLiteMapLoader();
+MapDatabase map = loader.load(Arrays.asList(param));
+```
+
+#6. Physical Format
+
+<p>Choose the physical format of the OpenLR location reference container<br>
+The following three formats are available in this package:</p>
+<ul>
+<li>binary: openlr.binary</li>
+<li>protobuf: openlr.proto</li>
+<li>xml: openlr.xml</li>
+</ul>
+
+<p>We are choosing binary format for the remaining part of the document</p>
+
+#5. Encoder
+
+<p>The code snippet given below shows how to encode a line location on the map in OpenLR binary format.</p>
+
+<p>you can find the default encoder properties file in openlr/encoder/src/main/resources/OpenLR-Encoder-Properties.xml
+</p>
+
+***In the example given below we are encoding a line location with connected lines (1,2,3,4)***
+
+```java
+List<Line>  testLocation = Arrays.asList(mapDatabaseAdapter.getLine(1),
+                                         mapDatabaseAdapter.getLine(2),
+                                         mapDatabaseAdapter.getLine(3),
+                                         mapDatabaseAdapter.getLine(4));
+Location location = LocationFactory.createLineLocation("Test location", testLocation,0,0);
+PhysicalEncoder physicalEncoder = new OpenLRBinaryEncoder();
+Configuration encoderConfig = OpenLRPropertiesReader.loadPropertiesFromFile(new File("OpenLR-Encoder-Properties.xml"));
+OpenLREncoderParameter params = new OpenLREncoderParameter.Builder().with(map).with(encoderConfig)
+                .with(Arrays.asList(physicalEncoder))
+                .buildParameter();
+OpenLREncoder encoder = new openlr.encoder.OpenLREncoder();
+LocationReferenceHolder locationReferenceHolder = encoder.encodeLocation(params, location);
+String locationReferenceBinary = ((ByteArray) physicalEncoder.encodeData(locationReferenceHolder.getRawLocationReferenceData()).getLocationReferenceData()).getBase64Data();
+```
+
+#6. Decoder
+
+<p>The code snippet given below shows how to decode an OpenLR binary location reference on to a map.</p>
+<p>you can find the default encoder properties file in /Users/babub/scratch/TomTom_GitRepo/openlr/decoder/src/main/resources/OpenLR-Decoder-Properties.xml</p>
+
+***In the example given below we are decoding OpenLR location reference binary string***
+
+```java
```

Comment:
Some whitespace between the relevant sections with some comments above them to say what you are doing at each point would be good.

Reply:
388c4431003b2ac034fc1c1f8c2fe6c86deaf599